src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -451,3 +456,52 @@ export const handleFormAuthRedirect = [
   }),
   _handleFormAuthRedirect,
 ] as RequestHandler[]
+
+/**
+ * Handler for validating the eServiceId of a given form
+ *
+ * @returns 200 with eserviceId validation result
+ * @returns 400 when there is an error on the authType of the form
+ * @returns 400 when the eServiceId of the form does not exist
+ * @returns 404 when form with given ID does not exist
+ * @returns 500 when the title of the fetched login page does not exist
+ * @returns 500 when database error occurs
+ * @returns 500 when the url for the login page of the form could not be generated
+ * @returns 502 when the login page for singpass could not be fetched
+ */
+export const handleValidateFormEsrvcId: RequestHandler<
+  { formId: string },
+  PublicFormAuthValidateEsrvcIdDto | ErrorDto
+> = (req, res) => {
+  const { formId } = req.params
+  return FormService.retrieveFormById(formId)
+    .andThen((form) => {
+      // NOTE: Because the check is based on parsing the html of the returned webpage,
+      // And because MyInfo login is beyond our control, we coerce MyInfo to SP.
+      // This is valid because a valid MyInfo eserviceId is also a valid SP eserviceId
+      if (form.authType === AuthType.MyInfo) {
```

Comment:
Suggest switch-case, what if the auth type is NIL?
is there also a regression? this used to only be called by SP/MyInfo forms, and not for CP forms since we cannot validate CP forms. 

Should we also check if that has changed, since CP is now logged in via SP?

Reply:
good catch on the first point! i'll maintain backwards compatibility for the `CP` one first and then raise an issue to investigate if it's possible!